### ISSUE TITLE
`--path` autodetection with multiple *.auto.tfvars files results in duplicate projects

### ISSUE THREAD

**Issue by JohnLyman** [infracost/infracost#2624]
If there are multiple `*.auto.tfvars` files in a directory, infracost autodetects multiple projects, resulting in duplicates.

For example, using [the example repo](https://github.com/infracost/example-terraform), if I run `infracost breakdown` without changing anything, it correctly detects a single project:
```
$ infracost breakdown --path .
Evaluating Terraform directory at .

...

 OVERALL TOTAL                                                                                                $680.14
──────────────────────────────────
2 cloud resources were detected:
∙ 2 were estimated, all of which include usage-based costs, see https://infracost.io/usage-file

┏━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━┳━━━━━━━━━━━━━━┓
┃ Project                                            ┃ Monthly cost ┃
┣━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━╋━━━━━━━━━━━━━━┫
┃ infracost/example-terraform                        ┃ $680         ┃
┗━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━┻━━━━━━━━━━━━━━┛
```
But if I create five `*auto.tfvars` files, it then detects five projects with five times as many cloud resources and five times the cost:
```
$ touch {1..5}.auto.tfvars
$ infracost breakdown --path .
Evaluating Terraform directory at .
Detected Terraform project at .
Detected Terraform project at .
Detected Terraform project at .
Detected Terraform project at .
Detected Terraform project at .

...

 OVERALL TOTAL                                                                                              $3,400.70
──────────────────────────────────
10 cloud resources were detected:
∙ 10 were estimated, all of which include usage-based costs, see https://infracost.io/usage-file

┏━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━┳━━━━━━━━━━━━━━┓
┃ Project                                            ┃ Monthly cost ┃
┣━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━╋━━━━━━━━━━━━━━┫
┃ infracost/example-terraform-1.auto                 ┃ $680         ┃
┃ infracost/example-terraform-2.auto                 ┃ $680         ┃
┃ infracost/example-terraform-3.auto                 ┃ $680         ┃
┃ infracost/example-terraform-4.auto                 ┃ $680         ┃
┃ infracost/example-terraform-5.auto                 ┃ $680         ┃
┗━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━┻━━━━━━━━━━━━━━┛
```

Using a simple config file with `path: .` correctly results in only a single project, even with multiple `*.auto.tfvars` files:
```
$ echo -e "version: 0.1\nprojects:\n  - path: ." > infracost.yml
$ ls *.auto.tfvars
1.auto.tfvars  2.auto.tfvars  3.auto.tfvars  4.auto.tfvars  5.auto.tfvars
$ infracost breakdown --config-file infracost.yml
Evaluating Terraform directory at .

...

OVERALL TOTAL                                                                                                $680.14
──────────────────────────────────
2 cloud resources were detected:
∙ 2 were estimated, all of which include usage-based costs, see https://infracost.io/usage-file

┏━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━┳━━━━━━━━━━━━━━┓
┃ Project                                            ┃ Monthly cost ┃
┣━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━╋━━━━━━━━━━━━━━┫
┃ infracost/example-terraform                        ┃ $680         ┃
┗━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━┻━━━━━━━━━━━━━━┛
```

I also noticed duplicate projects detected with multiple `*.auto.tfvars.json` files, or a mix of `*.auto.tfvars.json` and `*.auto.tfvars`.  

I'm using infracost v0.10.28.

**Comment by alikhajeh1** [comment]
Thanks for the awesomely detailed issue @JohnLyman! I'm curious to know what you think the expected behavior could be in this case, should we pass all of the .auto.tfvars files into the same project? as that's what TF would do, to load them all in.

**Comment by JohnLyman** [comment]
No problem.  Yes, I think it should load them all in.  I wouldn't expect to see my costs increase just because I added another `*.auto.tfvars` file (and no other cloud resources.)

**Comment by JohnLyman** [comment]
Great.  That was a quick turnaround, thanks @hugorut !

**Comment by aliscott** [comment]
@JohnLyman this is now released in [Infracost v0.10.29](https://github.com/infracost/infracost/releases/tag/v0.10.29).
